### PR TITLE
Add dev config with default jar task on it, to aid with depending on …

### PR DESCRIPTION
**Pull request by modmuss50**
…sub projects.

Dpeneding on a sub project is now just `project(path: ":name", configuration: "namedElements")`.

### CODE REVIEW

**Review by shedaniel — APPROVED** [review]

File: src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java [hunk 1]
```diff
@@ -57,6 +57,7 @@ public static void setupConfigurations(Project project) {
 		extension.createLazyConfiguration(Constants.Configurations.MINECRAFT).configure(configuration -> configuration.setTransitive(false));
 		extension.createLazyConfiguration(Constants.Configurations.INCLUDE).configure(configuration -> configuration.setTransitive(false)); // Dont get transitive deps
 		extension.createLazyConfiguration(Constants.Configurations.MAPPING_CONSTANTS);
+		extension.createLazyConfiguration(Constants.Configurations.DEV);
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java]: Should be declared `canBeResolved = false; canBeConsumed = true`

**Review by Juuxel — CHANGES_REQUESTED** [review]

**Comment by Juuxel** [comment]
`dev` is a bit generic of a name, so I wonder if it'll conflict a lot with existing configurations declared by users? You could always go with `devElements` to mirror `apiElements`/`runtimeElements`. But a good change nonetheless :rocket:

**Comment by modmuss50** [comment]
I was unsure on it conflicting, `named`, `namedOutputs` or `namedElements` might make more sense?

**Comment by Juuxel** [comment]
`namedOutputs` and `namedElements` would both be good imo. "outputs" is clear to the user, but "elements" does follow the gradle conventions, which might be good as well.

**Review by Juuxel — APPROVED** [review]

File: src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java [hunk 2]
```diff
@@ -147,6 +151,9 @@ public static void configureCompile(Project p) {
 			}
 		});
 
+		// Add the "dev" jar to the "dev" configuration
```
Comment (Juuxel) [inline src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java]:
```suggestion
		// Add the "dev" jar to the "namedElements" configuration
```

**Pull request merged**